src/post.ts

```diff
@@ -1,16 +1,31 @@
 import { setFailed } from '@actions/core'
-import { mkdirP, mv } from '@actions/io'
+import { mkdirP, mv, cp } from '@actions/io'
+import { spawn } from 'child_process'
+import { promisify } from 'util'
 
 import { getVars } from './lib/getVars'
 import { isErrorLike } from './lib/isErrorLike'
 import log from './lib/log'
 
+const rsync = promisify(spawn)
+
 async function post(): Promise<void> {
   try {
-    const { cacheDir, targetPath, cachePath } = getVars()
+    const { cacheDir, targetPath, cachePath, options } = getVars()
 
     await mkdirP(cacheDir)
-    await mv(targetPath, cachePath, { force: true })
+    if (options.copyStrategy === 'move') {
+      await mv(targetPath, cachePath, { force: true })
+    } else if (options.copyStrategy === 'copy') {
+      await cp(targetPath, cachePath, { copySourceDirectory: true, force: true, recursive: true })
+    } else if (options.copyStrategy === 'rsync') {
+      await rsync('rsync', ['-avqR', targetPath, cacheDir], { stdio: 'inherit' })
+    } else {
+      setFailed(`Unknown copy strategy ${options.copyStrategy}`)
```

Comment:
Better to throw an error.

```suggestion
      throw new TypeError(`Unknown strategy: ${options.strategy}`)
```